examples/gym/molecule_gym/molecule_graph_problem.py

```diff
@@ -74,15 +75,24 @@ def make_observation(self, state: MoleculeState) -> {str: np.ndarray}:
     def get_initial_state(self) -> MoleculeState:
         return MoleculeState(rdkit.Chem.MolFromSmiles('C'), self.builder)
 
-    def step(self, state: MoleculeState) -> (float, bool, dict):
+    @abstractmethod
+    def reward(self, state: MoleculeState) -> float:
+        pass
+
+    def step(self, state: MoleculeState) -> Tuple[float, bool, dict]:
         is_terminal = len(state.get_next_actions()) == 0
         if state.forced_terminal:
-            r = qed(state.molecule)
+            r = self.reward(state)
             print(r)
             return r, is_terminal, {'forced_terminal': True, 'smiles': state.smiles}
             # return qed(state.molecule), is_terminal, {'forced_terminal': True, 'smiles': state.smiles}
         return 0.0, is_terminal, {'forced_terminal': False, 'smiles': state.smiles}
 
     @property
-    def invalid_action_result(self) -> (float, bool, {}):
+    def invalid_action_result(self) -> Tuple[float, bool, Dict]:
         return -1.0, True, {}
+
+
+class QEDGraphProblem(MoleculeGraphProblem):
+    def reward(self, state: MoleculeState) -> float:
+        return qed(state.molecule)
```

Comment:
I think if this works we should rethink how we want this whole package structured. Tracking down these classes is a bit jumbled currently